demos/data/scripts/generator/PersonGen.py

```diff
@@ -0,0 +1,438 @@
+                                        maximum_age=max_age
+                                        ),
+            }
+            record["email_address"] = record["first_name"] + record["last_name"] + str(self.random.randint(0, 999)) + self.random.choice(self.domains)
+            records.append(record)
+        
+        df = pd.DataFrame(records)
+        return df
+
+    def get_address(self) -> dict:
+        # Placeholder for your existing `self.address()` method
+        return real_random_address.RandomAddress()
+
+    def get_address_by_state(self, state: str) -> dict:
+        # Placeholder for generating an address by state
+        # Implement actual functionality here
+        return real_random_address.real_random_address_by_state(state)
+
+
+    def get_address_by_postal_code(self, postal_code: str) -> dict:
+        # Placeholder for generating an address by postal code
+        return real_random_address.real_random_address_by_postal_code(postal_code)
+
+    def generate_addresses(
+            self,
+            num_records: int = 100,
+            start_date: str = "-30y",
+            end_date: str = "today",
+            state: str = None,
+            postal_code: str = None
+            ) -> pd.DataFrame:
+        """
+        Generate addresses for a set of people, simulating a history of addresses.
+        :param num_records: Number of addresses to generate.
+        :param start_date: Start date for address history.
+        :param end_date: End date for address history.
+        :param state: Optional state to generate addresses for.
+        :param postal_code: Optional postal code to generate addresses for.
+        :return: DataFrame of addresses.
+        :raises ValueError: If both state and postal_code are provided.
+        """
+
+        if state and postal_code:
+            raise ValueError("Cannot specify both state and postal code. Please choose one.")
+
+        records = []
+        for _ in range(num_records):
+            if state:
+                address = self.get_address_by_state(state)
+            elif postal_code:
+                address = self.get_address_by_postal_code(postal_code)
+            else:
+                address = self.get_address()
+
+            record = {
+                "address1": address.get('address1', ''),
+                "address2": address.get('address2', ''),
+                "city": address.get("city", "Unknown City"),
+                "date": self.fake.date_between(start_date=start_date, end_date=end_date),
+                "state": address.get("state", "Unknown State"),
+                "zip": address.get("postalCode", "Unknown PostalCode"),
+                "lat": address.get("coordinates", {}).get("lat", 0.0),
+                "lon": address.get("coordinates", {}).get("lng", 0.0)
+            }
+            records.append(record)
+        df = pd.DataFrame(records)
+        return df
+
+    def generate_call_logs(
```

Comment:
should there be a notion of tracked phone numbers?

- most foolks slooowly rotate the main phone
- some folks quickly rotate their burners, and are primarily used for calling burners? something  like that

Reply:
(and burners are / aren't associated with people, e.g., sometimes known, sometimes not?)

Reply:
Yea was thinking maybe I do something similar to the whereabouts where it has a date tied to the person at that address

Reply:
yeah, similar to the social network, i think either:

- prepick a distribution/network/etc ahead of time, and follow that
- do a per-person/community timeline 'simulation'

i think we want to guarantee that burners are active only for a period of time before retiring, so i can imagine some sort of simulation:

```
person_to_burners : Dict[PersonId, List[BurnerId]]
person_to_main_phone : Dict[PersonId, PhoneId]
for tick:
  person = pick(person_to_burners.keys())
  ACTION_CALL_BURNER
  ACTION_CALL_MAIN
  ACTION_DESTROY_BURNER
  ACTION_NEW_BURNER
  ACTION_REPLACE_PHONE
  
  switch random():
     case ACTION_CALL_BURNER:
       ...
```